gqcp/include/FockSpace/WaveFunction/WaveFunctionSelection.hpp

```diff
@@ -0,0 +1,76 @@
+// This file is part of GQCG-gqcp.
+// 
+// Copyright (C) 2017-2019  the GQCG developers
+// 
+// GQCG-gqcp is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+// 
+// GQCG-gqcp is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Lesser General Public License for more details.
+// 
+// You should have received a copy of the GNU Lesser General Public License
+// along with GQCG-gqcp.  If not, see <http://www.gnu.org/licenses/>.
+// 
+#pragma once
+
+
+#include "Basis/TransformationMatrix.hpp"
+#include "FockSpace/SelectedFockSpace.hpp"
+#include "FockSpace/WaveFunction/WaveFunction.hpp"
+#include "Mathematical/Representation/Matrix.hpp"
+
+
+
+namespace GQCP {
+
+
+/**
+ *  A class that represents a selection of a wave function
+ */
+class WaveFunctionSelection {
+private:
+    std::vector<Configuration> configurations;
+    VectorX<double> coefficients;  // the coefficients for the ONV configurations
+
+public:
+    // CONSTRUCTORS
+    WaveFunctionSelection() = default;
+
+    /**
+     *  Construct a selected wave function
+     *
+     *  @param fock_space           the selected Fock space in which the wave function 'lives'
+     *  @param coefficients         the expansion coefficients
+     */
+    WaveFunctionSelection(const SelectedFockSpace& fock_space, const VectorX<double>& coefficients);
+
+
+    /**
+     *  Construct a selected wave function by selecting the configurations with the highest contribution from a different wave function
+     *
+     *  @param wave_function                           the wave_function to be selected from
+     *  @param number_of_largest_contributions         the amount of selections made
+     */
+    WaveFunctionSelection(const WaveFunction& wave_function, size_t number_of_largest_contributions);
+
+
+    // PUBLIC METHODS
+    /**
+     *  Print this wave function to an output in the GAMESS-US Expansion format
+     *
+     *  @param output_stream        the stream used for outputting
+     */
+    void printGamessExpansion(std::ostream& output_stream = std::cout) const;
```

Comment:
Could you change this API to return a string instead of effectively printing it?

```cpp
std::string asGAMESSExpansion() const;
```

Reply:
The issue is how you chose the format of the coefficients in terms of precision and notation type? if you return it as a string instead of a stream?

Reply:
Good catch! I didn't think of that.